numpyro/distributions/multivariate.py

```diff
@@ -7,27 +7,80 @@
 # All rights reserved.
 
 import jax.numpy as np
+from jax import lax
+from jax.experimental.stax import softmax
+from jax.numpy.lax_numpy import _promote_dtypes
 from jax.scipy.special import digamma, gammaln
 
 from numpyro.distributions import constraints
-from numpyro.distributions.distribution import jax_mvcontinuous
-from numpyro.distributions.util import standard_gamma, xlogy
+from numpyro.distributions.discrete import binom
+from numpyro.distributions.distribution import jax_continuous, jax_discrete, jax_multivariate
+from numpyro.distributions.util import categorical_rvs, entr, multinomial_rvs, standard_gamma, xlogy
 
 
 def _lnB(alpha):
     return np.sum(gammaln(alpha), axis=-1) - gammaln(np.sum(alpha, axis=-1))
 
 
-class dirichlet_gen(jax_mvcontinuous):
-    # TODO: use dirichlet doc instead of the default one of rv_continuous
+# TODO: either use multivariate docs instead of the default one of
+# rv_continuous/rv_discrete
+
+class categorical_gen(jax_multivariate, jax_discrete):
```

Comment:
`jax_discrete` has a special __init__ which includes `is_logits` flag. In addition, because we don't have a common method for `logpdf` `logpmf` so by inheriting from `jax_discrete`, we just need to check `isinstance(dist, jax_discrete)`. Beside these convenience, there is no need for inheritance.